packages/main/src/RPA/Robocorp/utils.py

```diff
@@ -69,6 +69,24 @@ def log_more(message, *args, func=logging.debug):
         log_to_console(str(message) % args)
 
 
+def get_dot_value(source: Dict, key: str) -> Any:
+    """Returns the end value from `source` dictionary given `key` traversal."""
+    keys = key.split(".")
+    value = source
+    for _key in keys:
+        if not isinstance(value, dict):
+            return value
+        value = value.get(_key)
+    return value
+
+
+def set_dot_value(source: Dict, key: str, *, value: Any):
+    """Sets the end `value` into `source` dictionary given `key` destination."""
+    keys = key.rsplit(".", 1)
+    source = source if len(keys) == 1 else get_dot_value(source, keys[0])
```

Comment:
```suggestion
    if len(keys) > 1:
        source = get_dot_value(source, keys[0])
```
not a one liner, but might make it a bit more readable. What do you think?